src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -642,5 +674,84 @@ public static ClusterState readFrom(StreamInput in, @Nullable DiscoveryNode loca
             }
             return builder.build();
         }
+
+    }
+
+    public Diff diffs(ClusterState previousState) {
+        return new ClusterStateDiff(previousState, this);
+    }
+
+
+    private class ClusterStateDiff implements Diff {
+
+        private final long version;
```

Comment:
Anything that doesn't start with "previous" is "after". If I will just make toVersion and leave everything else with "to" prefix it will be inconsistent. If I will add "to" prefix everywhere it will be redundant.